CIP-dapp-bridge-rework/README.md

```diff
@@ -0,0 +1,1169 @@
+
+## Motivation: why is this CIP necessary?
+
+This CIP addresses several issues that complicate the development of the Cardano ecosystem with the current integration standard:
+
+1. **Complexity in Writing Client-Side Logic for DApps**: As they essentially need to implement their own wallet each time, including algorithms for coin selection, UTxO management, UTxO validation, TX building, etc. They shouldn't be doing this. There is already a separately written wallet that connects to their application for this purpose. <br><br>
+
+2. **Integration Difficulties**: This implies complicating the internal logic of wallets, adding endpoints for importing transactions, writing logic for checks/validation of foreign transactions, etc. This could be avoided if the transaction was created and processed natively by the wallet.<br><br>
+
+3. **Privacy Issues**: Currently, the wallet must pass all information to the DApp, including the UTxO set and full addresses on one hand, and trust the user's funds to the transaction building code inside the DApp on the other. This is not obvious to the user, and in case of problems, they will blame the wallet developers.<br><br>
+
+4. **Current Role of Wallets**: At present, the wallet is essentially just a synchronization point for the client side of the DApp with the blockchain. Yes, it generates user signatures, but, for example, if we use a HW-wallet, then the task of the SW-wallet is reduced to 'submit tx'. This is an incorrect design of communication.<br><br>
+
+5. **Scalable Complexity**: Currently, the integration standard implies that the complexity of the solution on the DApp side **must be** unjustifiably high even for elementary things that do not require knowledge of CBOR, CDDL, or an understanding of the difference between bech32 and base58.<br><br>
+
+6. **Reducing Dependencies**: Currently, all DApps developers **must** operate only with CBOR-CDDL entities, making them dependent on third-party libraries such as `cardano_serialization_lib`, and forcing them to integrate these libraries into their projects. This complicates the logic, increases the size of the loaded page, necessitates keeping up with updates, etc., thereby increasing the number of failure factors in the application's logic.<br><br>
+
+7. **Lack of abstraction**: The Cardano ecosystem is actively developing, especially in everything related to Plutus. DApps struggle to keep up with the innovations. Because of this, we encounter a heavy inconsistency in approaches. For example, this concerns everything related to collaterals at the moment. Cardano has long allowed making their use transparent and convenient for the user, but not all DApps support these approaches, forcing wallets to "reserve collateral UTxO" and somehow integrate this into UX. Moreover, the very fact that the DApp operates the collateral, and not the wallet (which only offers collateral but without guarantees that it will be used), makes the user vulnerable.<br><br>
```

Comment:
> makes the user vulnerable

No it does not, phase-2 validity is still checked by the wallet.

Reply:
In a perfect world everything is perfect. But as a software engineer I can say that in a real world we have to deal with imperfect software which can work in imperfect way. With each new link in the chain of responsibility, new potential points of failure arise. Each new link increases the number of potential attack vectors if there is lack of checking. The goal of this CIP is to make all operations for the wallet as **explicit as possible**. The wallet won't have to recheck every field of the transaction passed to it from the outside, if it assembled the transaction itself and processed every aspect of it 100%.

In this case, even in an imperfect world, we will reduce the likelihood of problems.